### ISSUE TITLE
Fix email addresses containing spaces

### ISSUE THREAD

**Issue by we-ai** [episphere/connect#1106]
Issue found:
- Some participant's emails (CID 869588347, maybe also other CID) contain spaces (" ") in the middle or at the end. Example formats found in prod data: `myemail2024@outlook.com `,  `My Email 2024@gmail.com`.  These email addresses (with spaces) don't pass email format checking, thus don't receive our notifications.

Things to do:
- Fix current data. We may need to check our data, and remove spaces in email addresses.
- Do more checks (spaces) and data cleanups in PWA for future email inputs.

@brotzmanmj @anthonypetersen Please comment.

**Comment by anthonypetersen** [comment]
I would say the most important piece is putting logic in the PWA to stop this from happening in the future. We could easily find an email regex out there that would clean this up.

Cleaning up the emails with leading or tailing spaces isn't too difficult, but the ones with spaces in the middle I worry we would make an incorrect assumptions sometimes by just removing the space.

**Comment by we-ai** [comment]
> I would say the most important piece is putting logic in the PWA to stop this from happening in the future. We could easily find an email regex out there that would clean this up.
> 
> Cleaning up the emails with leading or tailing spaces isn't too difficult, but the ones with spaces in the middle I worry we would make an incorrect assumptions sometimes by just removing the space.

Totally agreed.
Spaces in the middle are tricky, but we don't have other options aside from removing them. Since spaces are not allowed in the middle, even we consider them correct, sending emails to these addresses won't succeed.

**Comment by we-ai** [comment]
Another thing to add. Maybe a participant should've received multiple emails already, but never received an email because the email address contains space(s). Once the email address is fixed and without any restrictions, he/she may receive the blocked emails all at once, some emails (e.g. one-year anniversary, etc) might've passed due already thus not appropriate. 

So before fixing email addresses, we may need to add placeholder emails to block unnecessary emails to participants.

**Comment by brotzmanmj** [comment]
I agree the higher priority is to add something to the PWA to prevent them from adding bad email addresses. I will add to the list of User Profile changes that are planned for Fall 2024.

For correcting existing ones, I think we should test in dev or stage to see if they will actually get the past messages or not.

**Comment by we-ai** [comment]
Update on 9/3. When dry running "3mo QOL Survey Reminders" 2nd contact in prod, of the 16705 checked participant data entries, there were 5 email addresses that contained spaces and failed to pass email format check. Below are tokens:
- 7ed16c62-4b82-4989-a695-ba455dcc258c
- cf51364e-8972-4797-b8fa-3a75015e25c9
- cc8df35a-2327-4c86-a03d-986e7acebb2b
- cd440e8f-b22e-4352-9fee-7ffc197f288b
- 247dc68e-30c6-4d41-9888-b1c372d6e9be

**Comment by brotzmanmj** [comment]
Thanks @we-ai . We don't currently have a process in place to fix these, so we will assume these participants will not get the email. The SMS reminder should go out today so hopefully that will reach these participants.

**Comment by we-ai** [comment]
@brotzmanmj Yes, SMS messages will be sent to them today, since these 5 participants all have "can we text" (CID 646873644) value as "yes"(CID 353358909).

**Comment by we-ai** [comment]
On 9/6, when dry running "3mo QOL Survey Reminders" 3rd contact in prod, there were 5 email addresses that contained spaces and failed to pass email format check. Tokens are same as those found when dry running 2nd contact.

**Comment by Davinkjohnson** [comment]
This is being resolved by https://github.com/episphere/connect/issues/1190
